File: 01 - Assign Cookies/p1.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

void print(vector<int> arr){
    for( auto val : arr){
        cout<<val<<" ";
    }
    cout<<endl;
}

int findContentChildren(vector<int>& g, vector<int>& s) {

    sort(g.begin(),g.end());
    sort(s.begin(),s.end());

    int ans = 0;

    print(g);
    print(s);

    int i=0,j=0;
    int n = g.size();
    int m = s.size();

    while( i<n && j<m){
        if( s[j] >= g[i] ){
            ans++;
            i++;
        }
        j++;
    }

    return ans;
    
}

int main(){

    vector<int> g {10,9,8,7};
    vector<int> s {5,6,7,8};

    int ans = findContentChildren(g,s);
    cout<<"max assignment = "<<ans<<endl;

    return 0;
}

File: 01 - Assign Cookies/p1.py
def findContentChildren(g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """

        g.sort()
        s.sort()
        
        i=0
        j=0
        n=len(g)
        m=len(s)
        ans = 0

        while (i<n) and (j<m):
            if g[i] <= s[j]:
                ans += 1
                i += 1
            j += 1
        
        return ans


if __name__=='__main__':
    g = [1,2,3,1]
    s = [1,1]

    print("cookies assign = ", findContentChildren(g,s))